tests/test_algorithms.py

```diff
@@ -0,0 +1,327 @@
+"""
+This program has been developed by students from the bachelor Computer Science at
+Utrecht University within the Software Project course.
+© Copyright Utrecht University (Department of Information and Computing Sciences)
+"""
+
+import math
+import pytest
+
+import numpy as np
+import pandas as pd
+
+from src.fairreckitlib.core.apis import ELLIOT_API
+from src.fairreckitlib.core.config_constants import KEY_RATED_ITEMS_FILTER
+from src.fairreckitlib.core.config_constants import TYPE_PREDICTION, TYPE_RECOMMENDATION
+from src.fairreckitlib.core.factories import Factory, GroupFactory
+from src.fairreckitlib.data.set.dataset_registry import DataRegistry
+from src.fairreckitlib.model.algorithms.base_algorithm import BaseAlgorithm
+from src.fairreckitlib.model.algorithms.base_predictor import BasePredictor
+from src.fairreckitlib.model.algorithms.base_recommender import BaseRecommender
+from src.fairreckitlib.model.algorithms.lenskit import lenskit_algorithms
+from src.fairreckitlib.model.algorithms.surprise import surprise_algorithms
+from src.fairreckitlib.model.model_factory import create_model_factory
+
+data_registry = DataRegistry('tests/datasets')
+dataset = data_registry.get_set('ML-100K')
+sub_set = dataset.load_matrix_df().sort_values(by=['user'], axis=0)[:1000]
+
+model_factory = create_model_factory()
+NUM_THREADS = 1
+non_deterministic_algos = [lenskit_algorithms.RANDOM, surprise_algorithms.NORMAL_PREDICTOR]
+top_k = [1, 5, 10]
+
+algo_kwargs = {
+    'rating_scale': (dataset.get_matrix_info('rating_min'), dataset.get_matrix_info('rating_max')),
+    KEY_RATED_ITEMS_FILTER: True, # recommenders only
+    'num_threads': NUM_THREADS
+}
+
+
+def test_model_factory():
+    """Test algorithms and factories in the factory to be derived from the correct base class."""
+    assert isinstance(model_factory, GroupFactory), 'expected model group factory.'
+
+    assert bool(model_factory.get_factory(TYPE_PREDICTION)), 'missing prediction models.'
+    assert bool(model_factory.get_factory(TYPE_RECOMMENDATION)), 'missing recommender models.'
+
+    for _, model_type in enumerate(model_factory.get_available_names()):
```

Comment:
Why not for model_type in model_factory.get_available_names() tho?

Reply:
right, all those for loops could be simplified then. I have been working with enumerate too much

Reply:
I mean we can discuss it with the TA, because using enumerate in this manner consistently seems pretty fine to me too. That's why I didn't bring it up before :p

Reply:
I have had the pylinter complain about it in the past, so I have been using it too often ever since.
Actually removing enumerate did reduce one of the too many local variables.